### PR TITLE
WebXR: fix depth sensing for Oculus Quest v68+

### PR DESCRIPTION
**Pull request by Maksims**
This PR fixes Depth Sensing for Meta Quest and all WebXR AR sessions in v2 engine.

With a recent update of specs and introduction of a new texture format (R16UI), Quest v68+ started using that format. Also there was issues introduced with two PRs: #6773 (was not a full cleanup) and #6831 (defines should be used instead of code concatenation).

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/xr/constants.js [hunk 1]
```diff
@@ -213,7 +213,15 @@ export const XRDEPTHSENSINGUSAGE_GPU = 'gpu-optimized';
 export const XRDEPTHSENSINGFORMAT_L8A8 = 'luminance-alpha';
 
 /**
- * Float 32 - indicates that depth sensing preferred raw data format is Float 32.
+ * Unsigned Short - indicates that depth sensing preferred raw data format as Unsigned Short 16 bit.
```
Comment (willeastcott) [inline src/framework/xr/constants.js:216]:
```suggestion
 * Unsigned Short - indicates that depth sensing preferred raw data format is Unsigned Short (16 bit).
```

File: src/framework/xr/constants.js [hunk 2]
```diff
@@ -213,7 +213,15 @@ export const XRDEPTHSENSINGUSAGE_GPU = 'gpu-optimized';
...
+ *
+ * @type {string}
+ * @category XR
+ */
+export const XRDEPTHSENSINGFORMAT_R16U = 'unsigned-short';
+
+/**
+ * Float 32 - indicates that depth sensing preferred raw data format is Float 32 bit.
```
Comment (willeastcott) [inline src/framework/xr/constants.js:224]:
```suggestion
 * Float 32 - indicates that depth sensing preferred raw data format is Float (32 bit).
```

**Comment by mvaligursky** [comment]
So this is for engine v2. I wonder if this fix would work for engine v1 and should be merged there too?

**Review by willeastcott — APPROVED** [review]

**Review by mvaligursky — APPROVED** [review]

File: examples/src/examples/xr/ar-camera-depth.example.mjs [hunk 3]
```diff
@@ -126,9 +118,8 @@ const updateShader = (array, float) => {
     const key = 'textureDepthSensing_' + array + float;
     let frag = fragShader;
```
Comment (willeastcott) [inline examples/src/examples/xr/ar-camera-depth.example.mjs:119]: Delete this line...

Comment (willeastcott) [reply examples/src/examples/xr/ar-camera-depth.example.mjs:119]: ...and set `fragShader` in `materialDepth.shaderDesc` instead.

**Comment by Maksims** [comment]
Actually, this needs to be cherry-picked to v1 too. As currently it crashes without this patch.

**Comment by mvaligursky** [comment]
Unfortunately it cannot be simply cherry-picked as it uses some v2 API, for example
```
materialDepth.setDefine('XRDEPTH_ARRAY', true);
```

**Pull request merged**